libkbfs/block_retrieval_queue.go

```diff
@@ -276,17 +282,22 @@ func (brq *blockRetrievalQueue) Shutdown() {
 }
 
 // TogglePrefetcher allows upstream components to turn the prefetcher on or off
-func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) error {
-	brq.prefetchMtx.Lock()
-	defer brq.prefetchMtx.Unlock()
+func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) (err error) {
+	var shutdownCh <-chan struct{}
+	func() {
+		brq.prefetchMtx.Lock()
+		defer brq.prefetchMtx.Unlock()
+		shutdownCh = brq.prefetcher.Shutdown()
```

Comment:
`blockRetrievalQueue.Shutdown` explicitly checks for a `nil` prefetcher (and old code now removed did too), so I assumed it was a possible state.